packages/ember-data/lib/system/model/model.js

```diff
@@ -295,7 +295,7 @@ var Model = Ember.Object.extend(Ember.Evented, {
     @property id
     @type {String}
   */
-  id: null,
+  id: Ember.computed.alias('_internalModel.id'),
```

Comment:
I think it would work for setting the id as well if we also implemented an ES5 setter.